# Programming Assignment: Выбор семейства распределений в наивном байесе#

Загрузите датасеты digits и breast_cancer из sklearn.datasets. Выведите несколько строчек из обучающих выборок и посмотрите на признаки.

In [1]:
from sklearn import model_selection, datasets, naive_bayes

can = datasets.load_breast_cancer()
dg = datasets.load_digits()

print('Breast_cencer data:')
print('Data:', can.data[:5,:5])
print('Target:', can.target[:5])

print('Digits data:')
print('Data:', dg.data[:5,:5])
print('Target:', dg.target[:5])

Breast_cencer data:
Data: [[1.799e+01 1.038e+01 1.228e+02 1.001e+03 1.184e-01]
 [2.057e+01 1.777e+01 1.329e+02 1.326e+03 8.474e-02]
 [1.969e+01 2.125e+01 1.300e+02 1.203e+03 1.096e-01]
 [1.142e+01 2.038e+01 7.758e+01 3.861e+02 1.425e-01]
 [2.029e+01 1.434e+01 1.351e+02 1.297e+03 1.003e-01]]
Target: [0 0 0 0 0]
Digits data:
Data: [[ 0.  0.  5. 13.  9.]
 [ 0.  0.  0. 12. 13.]
 [ 0.  0.  0.  4. 15.]
 [ 0.  0.  7. 15. 13.]
 [ 0.  0.  0.  1. 11.]]
Target: [0 1 2 3 4]


С помощью sklearn.cross_validation.cross_val_score c настройками по умолчанию и вызова метода mean() у возвращаемого этой функцией numpy.ndarray, сравните качество работы наивных байесовских классификаторов на этих двух датасетах. Для сравнения предлагается использовать BernoulliNB, MultinomialNB и GaussianNB.

__BernoulliNB__

In [2]:
clf = naive_bayes.BernoulliNB()
clf.fit(can.data, can.target)
print('Scorer (Breast_cencer):', round(model_selection.cross_val_score(clf, can.data, can.target).mean(),8))
clf.fit(dg.data, dg.target)
print('Scorer (Digits):', round(model_selection.cross_val_score(clf, dg.data, dg.target).mean(),8))

Scorer (Breast_cencer): 0.6274181
Scorer (Digits): 0.82417363


__MultinomialNB__

In [3]:
clf = naive_bayes.MultinomialNB()
clf.fit(can.data, can.target)
print('Scorer (Breast_cencer):', round(model_selection.cross_val_score(clf, can.data, can.target).mean(),8))
clf.fit(dg.data, dg.target)
print('Scorer (Digits):', round(model_selection.cross_val_score(clf, dg.data, dg.target).mean(),8))

Scorer (Breast_cencer): 0.89632045
Scorer (Digits): 0.87034974


__GaussianNB__

In [4]:
clf = naive_bayes.GaussianNB()
clf.fit(can.data, can.target)
print('Scorer (Breast_cencer):', round(model_selection.cross_val_score(clf, can.data, can.target).mean(),8))
clf.fit(dg.data, dg.target)
print('Scorer (Digits):', round(model_selection.cross_val_score(clf, dg.data, dg.target).mean(),8))

Scorer (Breast_cencer): 0.93851886
Scorer (Digits): 0.8069282


__Вопрос 1__

Каким получилось максимальное качество классификации на датасете breast_cancer?

0.93851886

__Вопрос 2__

Каким получилось максимальное качество классификации на датасете digits?

0.87034974

Вопрос 3

Выберите верные утверждения и запишите их номера через пробел (в порядке возрастания номера):

1) На вещественных признаках лучше всего сработал наивный байесовский классификатор с распределением Бернулли

2) На вещественных признаках лучше всего сработал наивный байесовский классификатор с мультиномиальным распределением

__3) Мультиномиальное распределение лучше показало себя на выборке с целыми неотрицательными значениями признаков__

__4) На вещественных признаках лучше всего сработало нормальное распределение__

3 4